_source/_posts/2024-02-29-net-scim.md

```diff
@@ -0,0 +1,1590 @@
+---
+layout: blog_post
+title: "How to Manage User Lifecycle with .NET and SCIM"
+author: ram-gandhi
+by: contractor
+communities: [.net]
+description: "Manage your user's lifecycles using .NET, SCIM, and an Identity Provider such as Okta."
+tags: [scim,dotnet]
+tweets:
+- ""
+- ""
+- ""
+image: blog/net-scim/social.jpg
+type: awareness
+github: https://github.com/oktadev/okta-net-scim-example
+---
+
+Several challenges exist in provisioning users for a growing company across systems and applications. It can be time-consuming, error-prone, challenging to sync users across domains, and potentially a security risk to keep track of existing profiles. Fortunately, a protocol called SCIM (System for Cross-domain Identity Management) standardizes user information across systems, making it possible to sync users. Also, combined with a SCIM-compliant Identity Provider (IdP), it securely automates common user lifecycle operations.
+In this tutorial, I'll walk you through building a .NET SCIM server with the CRUD (Create, Read, Update, Delete) Restful API endpoints outlined in the SCIM protocol. We'll then test the server using Swagger User Interface before integrating it with Okta, a SCIM compliant IdP. 
+
+## Use SCIM to manage users
+[SCIM](https://datatracker.ietf.org/doc/html/rfc7644) is an HTTP-based protocol defining standard endpoints and schemas to overcome complexity when implementing user lifecycle management operations across different systems.
+
+## Components of a SCIM integration
+There are two primary components for a SCIM integration. The first is the *SCIM server*, which is a server that has implemented the endpoints as described in the SCIM spec—for example, User endpoints, Group endpoints, Schemas endpoints, etc. The applications team typically develops these endpoints, which leaves the IdP to manage the user profiles. This blog will detail creating a sample SCIM server using .NET. The other component is the *SCIM client*, which makes SCIM-compliant HTTP calls to a SCIM server to exchange user profile information. Okta (the IdP), in our use case, will act as a SCIM client and call the .NET SCIM server we'll build using this tutorial.
```

Comment:
```suggestion
There are two primary components for a SCIM integration. The first is the *SCIM server*, which is a server that has implemented the endpoints as described in the SCIM spec — for example, User endpoints, Group endpoints, Schemas endpoints, etc. The applications team typically develops these endpoints, which leaves the IdP to manage the user profiles. This blog will detail creating a sample SCIM server using .NET. The other component is the *SCIM client*, which makes SCIM-compliant HTTP calls to a SCIM server to exchange user profile information. Okta (the IdP), in our use case, will act as a SCIM client and call the .NET SCIM server we'll build using this tutorial.
```

Per style guidelines

Reply:
em dash is longer than a dash -- hmmm TIL!